src/registries/menu_items_registry.ts

```diff
@@ -24,89 +25,117 @@ import { SpreadsheetChildEnv } from "../types/env";
  *    NB: a separator defined on the last item is not displayed !
  *
  */
-export interface MenuItem {
+interface MenuItemSpec {
   name: string | ((env: SpreadsheetChildEnv) => string);
   description?: string;
-  sequence: number;
+  sequence?: number;
   id?: string;
   isVisible?: (env: SpreadsheetChildEnv) => boolean;
   isEnabled?: (env: SpreadsheetChildEnv) => boolean;
   isReadonlyAllowed?: boolean;
   action?: (env: SpreadsheetChildEnv) => unknown;
-  children?: menuChildren;
+  children?: MenuChildren;
   separator?: boolean;
   icon?: string;
   textColor?: string;
 }
 
-export type FullMenuItem = Required<MenuItem>;
-
-type MenuItemsBuilder = (env: SpreadsheetChildEnv) => FullMenuItem[];
-type menuChildren = (FullMenuItem | MenuItemsBuilder)[];
+export interface MenuItem {
+  id: string;
+  sequence: number;
+  name: (env: SpreadsheetChildEnv) => string;
+  isVisible: (env: SpreadsheetChildEnv) => boolean;
+  isEnabled: (env: SpreadsheetChildEnv) => boolean;
+  isReadonlyAllowed: boolean;
+  action?: (env: SpreadsheetChildEnv) => unknown;
+  children: (env: SpreadsheetChildEnv) => MenuItem[];
+  separator: boolean;
```

Comment:
Is it the time to find a better solution for the separator ?
For exemple, instead of `separator`, add a property `group`, and keep the menu items grouped and add a separator after each group ?

Reply:
In another PR ? :roll_eyes: 
Do you about something like adding `group: "add_row"`, `group: "add_col"` on each menu item, then group them based on the group property.
It begs some questions about the `sequence` property. How would it affect order across groups ?

Reply:
I don't know exactly how we could order the different groups, but it's the idea.
Anyway, it will be done in another PR.